src/ReactWrapperComponent.jsx

```diff
@@ -71,12 +71,18 @@ export default function createWrapperComponent(node, options = {}) {
     },
   };
 
-  if (options.context && node.type.contextTypes) {
+  if (options.context && (node.type.contextTypes || options.childContextTypes)) {
     // For full rendering, we are using this wrapper component to provide context if it is
-    // specified in both the options AND the child component defines `contextTypes` statically.
+    // specified in both the options AND the child component defines `contextTypes` statically
+    // OR the merged context types for all children (the node component or deeper children) are
+    // specified in options parameter under childContextTypes.
     // In that case, we define both a `getChildContext()` function and a `childContextTypes` prop.
+    let childContextTypes = node.type.contextTypes;
+    if (options.childContextTypes) {
+      childContextTypes = options.childContextTypes;
```

Comment:
Yeah, it's probably better. I'll update this, however, I think there should be better automatic fix to this issue.